generator/template.go

```diff
@@ -195,6 +195,97 @@ function getNotifyEntityArrivalSink<T>(notifyCallback: NotifyStreamEntityArrival
     }
   })
 }
+
+type RequestPayload = Record<string, unknown>;
+type Primitive = string | boolean | number;
+
+/**
+ * Checks if given value is of a primitive type
+ * @param  {unknown} value
+ * @return {boolean}
+ */
+function isPrimitive(value: unknown): boolean {
+  return (
+    typeof value === "string" ||
+    typeof value === "number" ||
+    typeof value === "boolean"
+  );
+}
+
+/**
+ * Flattens a deeply nested request payload and returns an object
+ * with only primitive values and non-empty array of primitive values
+ * @param  {RequestPayload} requestPayload
+ * @param  {String} path
+ * @return {Record<string, Primitive | Array<Primitive>}
+ */
+function flattenRequestPayload<T extends RequestPayload>(
+  requestPayload: T,
+  path: string = ""
+): Record<string, Primitive | Array<Primitive>> {
+  return Object.keys(requestPayload).reduce(
+    (acc: T, key: string): T => {
+      const value = requestPayload[key];
+      const newPath = path ? [path, key].join(".") : key;
+
+      const isObject =
```

Comment:
`isObject` as per commented might not cover the edge cases.

Reply:
Thank you for your feedback and I understand your concern. I have pasted the final form of `isPlainObj.js` from `ramda adjunct` library. Let me know what you think of this. My comment is here
https://github.com/grpc-ecosystem/protoc-gen-grpc-gateway-ts/pull/11#discussion_r620745005